### PR TITLE
Misc changes around unix sockets

### PR DESCRIPTION
**Pull request by stevenengler**
These are some changes required before we can support fixed-size send buffers for unix dgram sockets.

### CODE REVIEW

File: src/test/pipe/test_pipe.rs [hunk 1]
```diff
@@ -439,8 +439,16 @@ fn test_read_after_write_close_with_empty_buffer() -> Result<(), String> {
             );
         });
 
-        // read fd should be readable
-        assert!(is_readable(read_fd).unwrap());
+        // read fd should be POLLHUP
+        assert_eq!(
+            test_utils::poll_status(read_fd, 0).unwrap(),
+            // shadow doesn't support POLLHUP
```
Comment (sporksmith) [inline src/test/pipe/test_pipe.rs:445]: Should there be an issue for this? (Linked here?)

**Review by sporksmith — APPROVED** [review]

File: src/test/pipe/test_pipe.rs (hunk 1 above)
Comment (stevenengler) [reply src/test/pipe/test_pipe.rs:445]: Added.

File: src/test/pipe/test_pipe.rs [hunk 2]
```diff
@@ -475,13 +483,21 @@ fn test_read_after_write_close_with_nonempty_buffer() -> Result<(), String> {
...
 
         // the write fd is closed, but there are still bytes remaining
         assert_eq!(nix::unistd::read(read_fd, &mut buf).unwrap(), 3);
 
-        // read fd should be readable
-        assert!(is_readable(read_fd).unwrap());
+        // read fd should be POLLHUP
+        assert_eq!(
+            test_utils::poll_status(read_fd, 0).unwrap(),
+            // shadow doesn't support POLLHUP
```
[the remark this answers is not in the document]
Comment (stevenengler) [reply src/test/pipe/test_pipe.rs:495]: Added.

File: src/test/pipe/test_pipe.rs [hunk 3]
```diff
@@ -514,22 +530,38 @@ fn test_write_after_read_close_with_full_buffer() -> Result<(), String> {
...
 
         // close the read fd
         nix::unistd::close(read_fd).unwrap();
 
-        // write fd should be writable
-        assert!(is_writable(write_fd).unwrap());
+        // write fd should be POLLERR
+        assert_eq!(
+            test_utils::poll_status(write_fd, 0).unwrap(),
+            // shadow doesn't support POLLERR
```
[the remark this answers is not in the document]
Comment (stevenengler) [reply src/test/pipe/test_pipe.rs:543]: Added.

File: src/test/pipe/test_pipe.rs [hunk 4]
```diff
@@ -514,22 +530,38 @@ fn test_write_after_read_close_with_full_buffer() -> Result<(), String> {
...
             nix::unistd::write(write_fd, &[1, 2, 3]),
             Err(nix::errno::Errno::EPIPE)
         );
 
-        // write fd should be writable
-        assert!(is_writable(write_fd).unwrap());
+        // write fd should be POLLERR
+        assert_eq!(
+            test_utils::poll_status(write_fd, 0).unwrap(),
+            // shadow doesn't support POLLERR
```
[the remark this answers is not in the document]
Comment (stevenengler) [reply src/test/pipe/test_pipe.rs:561]: Added.

File: src/main/host/descriptor/pipe.rs [hunk 5]
```diff
@@ -72,7 +72,7 @@ impl Pipe {
         self.has_open_file = val;
     }
 
-    pub fn max_size(&self) -> usize {
+    pub fn max_size(&self) -> u64 {
```
[the remark this answers is not in the document]
Comment (stevenengler) [reply src/main/host/descriptor/pipe.rs]: Oops yeah I don't know why I thought it used u64. I've changed them to usize instead.

**Pull request merged**